### PR TITLE
Generate PDF automatically on each push to master

### PR DESCRIPTION
**Pull request by w3bdesign**
It would be nice to have a PDF version of the cheat sheet as it is easier to read offline as well as on-the-go. 

I have added a Github action that generates a PDF of all of the markdown files in the repository on each push to master.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Docs to PDF
```
Comment (sapegin) [inline .github/workflows/main.yml]:
```suggestion
name: Generate PDF
```

We don't have any “docs” here.

And let’s rename the file to something more meaningful, like `pdf.yml`.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# This workflow is triggered on pushes to the master branch.
```
Comment (sapegin) [inline .github/workflows/main.yml]: This comment repeats the code.

```suggestion
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  converttopdf:
```
Comment (sapegin) [inline .github/workflows/main.yml]:
```suggestion
  pdf:
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    name: Create our PDF
```
Comment (sapegin) [inline .github/workflows/main.yml]:
```suggestion
    name: Create PDF
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Generate PDF for each folder in /contents
```
Comment (sapegin) [inline .github/workflows/main.yml]: We don't have `contents` folder here.

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: mkrakowitzer/actions-makepdfs@master
```
Comment (sapegin) [inline .github/workflows/main.yml]: Let's us a fixed version of the action to avoid its breaking in the future.

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          markdown_dir: 
+          output_dir: pdfen
```
Comment (sapegin) [inline .github/workflows/main.yml]: I'm not sure I understand the name of the folder...

File: .github/workflows/main.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+
```
Comment (sapegin) [inline .github/workflows/main.yml]:
```suggestion

```

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      # Upload PDF of Markdown files
```
Comment (sapegin) [inline .github/workflows/main.yml]: Upload where?

**Review by sapegin — CHANGES_REQUESTED** [review]
Thanks for the pull request!

Could you add a screenshot? How can one access the generated PDF file?

**Comment by elektronik2k5** [comment]
@w3bdesign thanks for your contribution! I just wanted to say that although @sapegin's comments may seem nit-picky, I don't consider them so. I support each and every one of them - despite the fact most of them aren't functinally required.

It is the small details that differentiate a great thing from an "ok" one. I too expect nothing short of the best from team members and contributors in my projects.

Again, thanks for the contribution! :bowing_man:

**Comment by w3bdesign** [comment]
> 
> 
> Thanks for the pull request!
> 
> Could you add a screenshot? How can one access the generated PDF file?

image

File: .github/workflows/main.yml (hunk 6 above)
Comment (w3bdesign) [reply .github/workflows/main.yml]: Fixed to mkrakowitzer/actions-makepdfs@v1.2

File: .github/workflows/main.yml (hunk 9 above)
Comment (w3bdesign) [reply .github/workflows/main.yml]: See screenshot

File: .github/workflows/main.yml (hunk 7 above)
Comment (w3bdesign) [reply .github/workflows/main.yml]: @sapegin 

I recommend that you read through the documentation at https://github.com/marketplace/actions/create-pdf

Comment (sapegin) [reply .github/workflows/main.yml]: I don't this it will explain the choice of names in this pull request 🤔

**Comment by sapegin** [comment]
This screenshot doesn't answer any of my questions 🤷‍♂️ 

How does the PDF look like? How could we link to it from the Readme?

**Comment by w3bdesign** [comment]
@sapegin 

Unfortunately I currently don't know how to directly link to the latest generated PDF as each run is unique for each push to the master branch, however, you should at least be able to do something like this:

`<a href="https://github.com/sapegin/jest-cheat-sheet/actions">https://github.com/sapegin/jest-cheat-sheet/actions</a>`

<a href="https://github.com/sapegin/jest-cheat-sheet/actions">https://github.com/sapegin/jest-cheat-sheet/actions</a>

I have attached a sample PDF below.

If you want to see how it works and looks like in action, feel free to checkout my fork here: https://github.com/w3bdesign/jest-cheat-sheet

[Readme.pdf](https://github.com/sapegin/jest-cheat-sheet/files/5337338/Readme.pdf)

**Comment by sapegin** [comment]
I think it's not worth it: 

1. The quality of the PDF is very low: a giant logo that takes full page, no syntax highlighting, no ability to expand collapsed sections...
2. Looks like we can't link it anyway.

**Pull request closed without merging**